tezos-sig-gating-contracts/src/lib/schemas.ts

```diff
@@ -104,18 +118,42 @@ type String0x = z.infer<typeof String0x>;
 // >;
 
 // Tx Auth Data SIgnature
+// This is the info that is signed by Nexera ID's Back end
 export const TezosTxAuthData = z.object({
   chainID: z.string(),
   userAddress: TezosAddress,
   nonce: z.number(),
   blockExpiration: z.number(),
   contractAddress: TezosContractAddress,
   functionCallName: TezosEntrypointName,
-  functionCallArgs: String0x,
-  publicKey: z.string(),
+  functionCallArgs: z.array(z.unknown()),
+  signerPublicKey: TezosPublicKey,
 });
 export type TezosTxAuthData = z.infer<typeof TezosTxAuthData>;
 
+export const TezosTxAuthInput = z.object({
+  chainID: z.string(),
+  contractAddress: TezosContractAddress,
+  functionName: TezosEntrypointName,
+  args: z.array(z.unknown()),
+  userAddress: TezosAddress,
+  blockExpiration: z.number().int().optional(),
+  nonce: z.number().int().optional(),
+});
+export type TezosTxAuthInput = z.infer<typeof TezosTxAuthInput>;
+
+// this is what is used in the contract call
+export const TezosTxCalldata = z.object({
+  userAddress: TezosAddress,
+  blockExpiration: z.number(),
+  contractAddress: TezosContractAddress,
+  functionCallArgsBytes: z.array(z.unknown()),
```

Comment:
so this should be z.string right?

Reply:
functionCallName is a string which start with a % followed by a lowcase char

Reply:
>functionCallName is a string which start with a % followed by a lowcase char

Would be nice if you could add a zod schema for this

But in this case I was talking about "functionCallArgsBytes", because the name implies that this is the bytes version

Reply:
yes , functionCallArgsBytes is z.string()